### ISSUE TITLE
Execution of the object doesn't run the method

### ISSUE THREAD

**Issue by hyeluri** [Enigmatis/graphql-java-annotations#11]
I'm trying to port over some code written in vanilla graphql-java to start using annotations.

I've got code that looks like this:

``` java
    @GraphQLName("VersionType")
    @GraphQLDescription("VersionType")
    private static class Version {
        @GraphQLField
        public @NotNull String version() {
            return "1.0.0";
        }
    }

    @SneakyThrows
    private static GraphQLObjectType getVersionType() {
        return GraphQLAnnotations.object(Version.class);
    }

    private static GraphQLObjectType queryType = newObject()
        .name("Query")
        .field(newFieldDefinition()
            .name("Version")
            .type(GraphQLString)
            .staticValue("1.0.0")
            .build()
        )
        ...
        .build()


    protected static GraphQLSchema schema = newSchema()
        .query(queryType)
        .mutation(mutationType)
        .build();

    public static GraphQL graphQL = new GraphQL(schema);
```

Now, the object builds correctly and the schema for the object type shows up. But when I try to do any query, I always get null back. The `String version()` method isn't even called.

I'm using the latest version of GraphQL-java (2.0.0) and everything is working fine in the other parts of the scheme written without annotations.

Thanks for the help in advance.

**Comment by yrashk** [comment]
Did you figure the source of the issue?

On Monday, 23 May 2016, Harish notifications@github.com wrote:

> Closed #11
> https://github.com/graphql-java/graphql-java-annotations/issues/11.
> 
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub
> https://github.com/graphql-java/graphql-java-annotations/issues/11#event-669751609

## 

Y.

**Comment by nmn** [comment]
``` java
    @GraphQLName("VersionType")
    @GraphQLDescription("VersionType")
    private static class Version {
        @GraphQLField
        public @NotNull String version() {
            return "1.0.0";
        }
    }

    @SneakyThrows
    private static GraphQLObjectType getVersionType() {
        return GraphQLAnnotations.object(Version.class);
    }

    private static GraphQLObjectType queryType = newObject()
        .name("Query")
        .field(newFieldDefinition()
            .name("version")
            .type(getVersionType())
            .build()
        )
        ...
        .build()


    protected static GraphQLSchema schema = newSchema()
        .query(queryType)
        .mutation(mutationType)
        .build();

    public static GraphQL graphQL = new GraphQL(schema);
```

That is the code we were trying to use. So we were getting an exception. When the method was being invoked, it was invoked on null or an object that was not of the class 'Version'. We figured that out after going through the code.

However, I  still don't understand how to use annotations to define the QueryType which has no environment.getSource value. If the classes need to match, it'll always be called on Null.
